liquibase-couchbase/src/test/java/system/change/MutateInQuerySystemTest.java

```diff
@@ -0,0 +1,90 @@
+package system.change;
+
+import com.couchbase.client.java.json.JsonObject;
+import com.couchbase.client.java.manager.query.DropPrimaryQueryIndexOptions;
+import common.operators.TestCollectionOperator;
+import liquibase.Liquibase;
+import liquibase.ext.couchbase.types.Document;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import system.LiquibaseSystemTest;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.couchbase.client.java.manager.query.DropPrimaryQueryIndexOptions.dropPrimaryQueryIndexOptions;
+import static common.constants.ChangeLogSampleFilePaths.MUTATE_QUERY_FILTER_IN_REPLACE_DOCUMENTS_TEST_XML;
+import static common.constants.ChangeLogSampleFilePaths.MUTATE_QUERY_FILTER_IN_REPLACE_DOCUMENT_TEST_XML;
+import static common.constants.TestConstants.TEST_COLLECTION_3;
+import static common.constants.TestConstants.TEST_SCOPE;
+import static common.matchers.CouchbaseCollectionAssert.assertThat;
+import static common.operators.TestCollectionOperator.createOneFieldJson;
+import static liquibase.ext.couchbase.types.Document.document;
+
+public class MutateInQuerySystemTest extends LiquibaseSystemTest {
+
+    private static TestCollectionOperator testCollectionOperator;
+
+    private static final JsonObject expectedReplaceContent = createOneFieldJson("newDocumentField", "newDocumentValue");
+    private static final String[] documentIds = {"replaceDocument1", "replaceDocument2", "replaceDocument3"};
+
+    @BeforeEach
+    void cleanUpd() {
+        for (String docId : documentIds) {
+            if (testCollectionOperator.docExists(docId)) {
+                testCollectionOperator.getCollection().remove(docId);
+            }
+        }
+    }
+
+    @AfterAll
+    static void cleanIndex() {
+        DropPrimaryQueryIndexOptions options = dropPrimaryQueryIndexOptions().ignoreIfNotExists(true);
+        testCollectionOperator.dropPrimaryIndex(options);
+    }
+
+    @BeforeAll
+    static void createPrimaryIndex() throws InterruptedException {
+        bucketOperator.createCollection(TEST_COLLECTION_3, TEST_SCOPE);
+        testCollectionOperator = bucketOperator.getCollectionOperator(TEST_COLLECTION_3, TEST_SCOPE);
+        TimeUnit.SECONDS.sleep(2L);
```

Comment:
Do we have here the previous problem if we don't use sleep?

Reply:
Yes,  sometimes we have unexpected result without that delay